src/CapabilityService.WebApi/Infrastructure/Persistence/CapabilityTransactionalDecorator.cs

```diff
@@ -172,6 +31,16 @@ public async Task<Capability> UpdateCapability(Guid capabilityId, string newName
 
         public Task<IEnumerable<Capability>> GetAllCapabilities() => _inner.GetAllCapabilities();
         public Task<Capability> GetCapability(Guid id) => _inner.GetCapability(id);
+        public async Task DeleteCapability(Guid id)
+        {
+            using (var transaction = await _dbContext.Database.BeginTransactionAsync())
+            {
+                await _inner.DeleteCapability(id);
+
+                await _dbContext.SaveChangesAsync();
+                transaction.Commit();
+            }
+        }
 
         public async Task<Capability> CreateCapability(string name, string description)
```

Comment:
We only call SaveChanges once in the scope of this transaction (same with Leave & Join)...

Reply:
I am not sure we can expect a TransactionalDecorator to know if the wrapped classes has preformed the SaveChanges or not